src/v4/http/index.ts

```diff
@@ -116,8 +116,11 @@ export class InteractionwithRequest implements V4InteractionwithRequest {
     private opts: PactV4Options
   ) {}
 
-  willRespondWith(status: number, builder?: V4ResponseBuilderFunc) {
-    this.interaction.withStatus(status);
+  willRespondWith(
+    status: number | Matcher<number>,
+    builder?: V4ResponseBuilderFunc
```

Comment:
Unrelated, but is the builder pattern the right thing to have in this API? It's not very idomatic.

Reply:
I definitely ummed and ahh'd about this. It could probably be a well defined type (or possibly set of types). The issue with using TypeScript types to represent different possible inputs, is that people still use JS and eventually put in garbage, so you need validation on top of it. 

The builder pattern doesn't allow it, and saves all of this additional validation code, whilst also still giving you a typed interface. 

But I'll revisit before it's no longer behind a flag. Thanks.